### DISCUSSION TITLE
Container service discovery

### DISCUSSION THREAD

**Discussion by SalomonHenke** [microsoft/aspire#6048] [Q&A]
Good morning

I can't seem to get service discovery with containers to work and haven't been able to find documentation on this.

I have an image which I am able to run via. the apphost:

```
var imageToPdf = builder.AddContainer(
    name: "pdfToImage", 
    image: "localhost/pdfimages", 
    tag: "v1.0")
    .WithHttpEndpoint(targetPort: 8080, port: 8080, name: "pdfToImage");

var apiService = builder.AddProject<Projects.AspireAndCachingTest_ApiService>("apiservice")
    .WithReference(cache)
    .WithReference(imageToPdf.GetEndpoint("pdfToImage"));
```

This starts the container as expected via podman
<img width="902" alt="image" src="https://github.com/user-attachments/assets/f5c672cf-79cf-4abe-b086-7ca4c3dd710b">

    
And I register an http client with 
    `builder.Services.AddHttpClient<PDFTOIMAGEClient>(
    static client => client.BaseAddress = new("https+http://pdfToImage"));`

but get a No such host is known. (pdftoimage:443)

What to do...

Small follow-up question that might be besides the aspire scope: How does this works when deploying this to say azure container apps.   Is Aspire able to have my image be part of the deployment process and do the bindings between services as it is able to do for projects, databases, caches and such? 

Help is appreciated!

**Comment by MiklosPathy** [comment]
I have a very similar problem. My example project, reproducing it:
https://github.com/MiklosPathy/Aspire.ServiceDiscovery.Problem

My question, on StackOverflow:
https://stackoverflow.com/questions/79405450/service-discovery-for-custom-container-in-net-aspire

Question is, what i am doing wrong? Other integrations seems to work, so it must be something more to this, than in the custom service example in the Aspire docs.

**Comment by davidfowl** [comment]
```
    .WithHttpEndpoint(targetPort: 8080, port: 8080, name: "pdfToImage");
```

Remove the name. Named endpoints are referenced differently from the default "http"/"https" endpoints. More details here https://learn.microsoft.com/en-us/dotnet/aspire/service-discovery/overview
